### PR TITLE
Start of Analyzer for Elyses Analytic Enchantments

### PR DESCRIPTION
**Pull request by pertrai1**
Have a PR open in website copy repo that adds the comments - https://github.com/exercism/website-copy/pull/2144

### CODE REVIEW

**Comment by SleeplessByte** [comment]
I think this is a _very_ solid base for that exercise!

**Comment by pertrai1** [comment]
@SleeplessByte - once this pull request is complete and merged in, I will start another analyzer. Thanks

File: src/analyzers/concept/elyses-analytic-enchantments/index.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    this.solution.readExemplar(options.inputDir)
+
+    if (this.solution.isExemplar) {
+      output.add(EXEMPLAR_SOLUTION())
+      output.finish()
+    }
+
+    if (this.solution.hasforEach || this.solution.hasFor) {
+      output.add(MUST_AVOID_IMPERATIVE_LOOP())
+      output.disapprove()
```
Comment (SleeplessByte) [inline src/analyzers/concept/elyses-analytic-enchantments/index.ts:63]: I _think_ `disapprove` is deprecated. `.finish()` is fine :)

File: src/analyzers/concept/elyses-analytic-enchantments/index.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+    this.solution = this.checkStructure(parsed.program, parsed.source, output)
+    this.solution.readExemplar(options.inputDir)
+
+    if (this.solution.isExemplar) {
+      output.add(EXEMPLAR_SOLUTION())
+      output.finish()
+    }
+
+    if (this.solution.hasforEach || this.solution.hasFor) {
```
Comment (SleeplessByte) [inline src/analyzers/concept/elyses-analytic-enchantments/index.ts:61]: Perhaps it makes sense to update this to `this.solution.hasImperativeLoop`? That way, if there are new ways/different ways to loop we want to check, it's a single method to update.

**Review by SleeplessByte — APPROVED** [review]
This looks great. Small nits, and of course the website-copy PR, but overall amazing job @pertrai1 ! Sorry I couldn't make time inbetween, but if you still want help, shout out!

**Comment by pertrai1** [comment]
Thank you @SleeplessByte. I have update from your feedback. I want to start another one if you have any in mind.

**Comment by SleeplessByte** [comment]
Try it out!

**Pull request merged**